packages/embedded-wallet/src/ui/slider/Slider.test.tsx

```diff
@@ -0,0 +1,129 @@
+import Enzyme, {mount, ReactWrapper} from 'enzyme';
+import Adapter from 'enzyme-adapter-react-16';
+import React from 'react';
+import {Slider, SliderProps} from './Slider';
+import css from './Slider.module.css';
+
+Enzyme.configure({adapter: new Adapter()});
+
+type MockSlider = {
+  sliderWrapper: ReactWrapper<SliderProps>;
+  minLabelElement: ReactWrapper;
+  sliderElement: ReactWrapper;
+  maxLabelElement: ReactWrapper;
+  valueLabelElement: ReactWrapper;
+};
+
+const mockSlider = (props?: Partial<SliderProps>): MockSlider => {
+  const sliderWrapper = mount<SliderProps>(
+    <Slider min={0} max={100} step={1} unit="ETH" {...props} />
+  );
+
+  return refreshSliderFrom({sliderWrapper} as MockSlider);
+};
+
+const refreshSliderFrom = (slider: MockSlider) => {
+  const {sliderWrapper} = slider;
+
+  sliderWrapper.update();
+
+  return {
+    sliderWrapper: slider.sliderWrapper,
+    minLabelElement: sliderWrapper.find(`label.${css.min}`),
+    sliderElement: sliderWrapper.find("input[type='range']"),
+    maxLabelElement: sliderWrapper.find(`label.${css.max}`),
+    valueLabelElement: sliderWrapper.find(`label.${css.value}`)
+  };
+};
+
+const triggerKeyOn = (component: MockSlider, element: keyof MockSlider, key: string) => {
+  component[element].simulate('keydown', {key});
+  return refreshSliderFrom(component);
+};
+
+describe('UI - Slider', () => {
+  let component: MockSlider;
+
+  beforeEach(() => {
+    component = mockSlider();
+  });
+
+  it('can be instantiated', () => {
+    expect(component.minLabelElement.exists()).toEqual(true);
```

Comment:
You could destructure here :-P

Reply:
![Maleficent says...](https://media.giphy.com/media/7oZdXTFa8P5qU/giphy.gif)

I could. But do I want to?
.
.
.
(I will, I will.)